pkg/base1/cockpit-react-terminal.jsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "react",
+    "base1/term",
+], function(React, Term) {
+
+"use strict";
+
+/*
+ * A terminal component that communicates over a cockpit channel.
+ *
+ * The only required property is 'channel', which must point to a cockpit
+ * stream channel.
+ *
+ * The size of the terminal defaults to 80 by 25. It can be changed with the
+ * 'rows' and 'cols' properties.
+ *
+ * If the 'onTitleChanged' callback property is set, it will be called whenever
+ * the title of the terminal changes.
+ */
+var Terminal = React.createClass({
+    propTypes: {
+        cols: React.PropTypes.number,
+        rows: React.PropTypes.number,
+        channel: React.PropTypes.object.isRequired,
+        onTitleChanged: React.PropTypes.func
+    },
+
+    getDefaultProps: function () {
+        return {
+            cols: 80,
+            rows: 25
```

Comment:
if you add a trailing comma on every row like this, possible future changes have potentially fewer side effects / # lines affected